File: insertionSort/insertionSort.py
sampleArr = [-6, 20, 8, -2, 4]

def main():
    insertionSort(sampleArr)
    print(sampleArr)

def insertionSort(arr):
    for i in range(1, len(arr)):
        didSwap = True
        sor = i - 1
        nti = i
        while(didSwap):
            if arr[nti] < arr[sor]:
                placeholder = arr[nti]
                arr[nti] = arr[sor]
                arr[sor] = placeholder
                sor -= 1
                nti -= 1
            else:
                didSwap = False

if __name__ == "__main__":
    main()

File: insertionSort/insertionSort.js
const sampleArr = [-6, 20, 8, -2, 4];

function insertionSort(arr){
    for(let i = 1; i < arr.length; i++){ //cur = current index
        let didSwap = true;
        let sor = i - 1;
        let nti = i;
        while(didSwap){
            if (arr[nti] < arr[sor]){
                let placeholder = arr[nti]
                arr[nti] = arr[sor]
                arr[sor] = placeholder
                sor--;
                nti--;
            } else {
                didSwap = false;
            }
        }
    }
}

insertionSort(sampleArr);
console.log(sampleArr);